src/idl_parser.cpp

```diff
@@ -914,8 +914,20 @@ CheckedError Parser::ParseTable(const StructDef &struct_def, std::string *value,
   size_t fieldn = 0;
   auto err = ParseTableDelimiters(fieldn, &struct_def,
                                   [&](const std::string &name) -> CheckedError {
-    auto field = struct_def.fields.Lookup(name);
-    if (!field) {
+  const char *schema_key = "$schema";
+  bool is_schema_key = true;
+  for (std::string::size_type i = 0; i < name.size(); ++i) {
```

Comment:
I have only found a specification on how to reference the schema for the actual schema from the schema file (what a sencence). Here [Understanding-json](https://spacetelescope.github.io/understanding-json-schema/reference/schema.html). So it seems that only `$schema` is used. 

I have not found a specification how ordinary json documents reference its schema.

But you are  right normal compare should be fine.